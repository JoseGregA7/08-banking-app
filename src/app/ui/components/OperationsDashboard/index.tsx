import AccountOptions from "../AccountOptions";
import AccountStatus from "../AccountStatus";
import './style.scss';

const OperationsDashboard = ({ accountInfo, handleWithdrawTypeChange, handleWithdrawalAmountChange, handleWithdrawalSubmit, state, selectedOption, handleOptionClick, handleClose, handleDepositTypeChange, handleDepositAmountChange, handleDepositSubmit, handleBuySubmit, handleBuyTypeChange, handleBuyAmountChange}: any) => {
    return (
        <div className="OperationsDashboard__wrapper">
            <AccountStatus accountInfo={accountInfo} />
            <AccountOptions selectedOption={selectedOption} handleOptionClick={handleOptionClick} handleClose={handleClose} handleWithdrawTypeChange={handleWithdrawTypeChange} handleWithdrawalAmountChange={handleWithdrawalAmountChange} handleWithdrawalSubmit={handleWithdrawalSubmit} state={state} handleDepositTypeChange={handleDepositTypeChange} handleDepositAmountChange={handleDepositAmountChange} handleDepositSubmit={handleDepositSubmit} handleBuySubmit={handleBuySubmit} handleBuyTypeChange={handleBuyTypeChange} handleBuyAmountChange={handleBuyAmountChange} />
        </div>
    )
}

export default OperationsDashboard