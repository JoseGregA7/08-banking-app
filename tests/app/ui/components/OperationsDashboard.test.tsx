import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import OperationsDashboard from '../../../../src/app/ui/components/OperationsDashboard';
import { describe, expect, test, vitest } from 'vitest';

describe('OperationsDashboard', () => {
  const mockHandleWithdrawTypeChange = vitest.fn();
  const mockHandleWithdrawalAmountChange = vitest.fn();
  const mockHandleWithdrawalSubmit = vitest.fn();
  const mockHandleDepositTypeChange = vitest.fn();
  const mockHandleDepositAmountChange = vitest.fn();
  const mockHandleDepositSubmit = vitest.fn();
  const mockHandleBuySubmit = vitest.fn();
  const mockHandleBuyTypeChange = vitest.fn();
  const mockHandleBuyAmountChange = vitest.fn();
  const mockHandleOptionClick = vitest.fn();
  const mockHandleClose = vitest.fn();

  const accountInfo = { balance: 1000, currency: 'USD' };
  const selectedOption = 'withdraw';
  const state = { isLoading: false };

  test('should render the component with AccountStatus and AccountOptions', () => {
    const { getByText, getByRole } = render(
      <OperationsDashboard 
        accountInfo={accountInfo}
        handleWithdrawTypeChange={mockHandleWithdrawTypeChange}
        handleWithdrawalAmountChange={mockHandleWithdrawalAmountChange}
        handleWithdrawalSubmit={mockHandleWithdrawalSubmit}
        state={state}
        selectedOption={selectedOption}
        handleOptionClick={mockHandleOptionClick}
        handleClose={mockHandleClose}
        handleDepositTypeChange={mockHandleDepositTypeChange}
        handleDepositAmountChange={mockHandleDepositAmountChange}
        handleDepositSubmit={mockHandleDepositSubmit}
        handleBuySubmit={mockHandleBuySubmit}
        handleBuyTypeChange={mockHandleBuyTypeChange}
        handleBuyAmountChange={mockHandleBuyAmountChange}
      />
    );
    expect(getByText(`Estado de la Cuenta`)).toBeInTheDocument();
    expect(getByText('Opciones de la Cuenta')).toBeInTheDocument(); 
  });

  test('should call handleOptionClick when an option is clicked', () => {
    const { getByText } = render(
      <OperationsDashboard 
        accountInfo={accountInfo}
        handleWithdrawTypeChange={mockHandleWithdrawTypeChange}
        handleWithdrawalAmountChange={mockHandleWithdrawalAmountChange}
        handleWithdrawalSubmit={mockHandleWithdrawalSubmit}
        state={state}
        selectedOption={selectedOption}
        handleOptionClick={mockHandleOptionClick}
        handleClose={mockHandleClose}
        handleDepositTypeChange={mockHandleDepositTypeChange}
        handleDepositAmountChange={mockHandleDepositAmountChange}
        handleDepositSubmit={mockHandleDepositSubmit}
        handleBuySubmit={mockHandleBuySubmit}
        handleBuyTypeChange={mockHandleBuyTypeChange}
        handleBuyAmountChange={mockHandleBuyAmountChange}
      />
    );
    fireEvent.click(getByText('DepÃ³sitos'));
    expect(mockHandleOptionClick).toHaveBeenCalledTimes(1);
  });

  test('should call handleClose when close button is clicked', () => {
    const { getByRole } = render(
      <OperationsDashboard 
        accountInfo={accountInfo}
        handleWithdrawTypeChange={mockHandleWithdrawTypeChange}
        handleWithdrawalAmountChange={mockHandleWithdrawalAmountChange}
        handleWithdrawalSubmit={mockHandleWithdrawalSubmit}
        state={state}
        selectedOption={selectedOption}
        handleOptionClick={mockHandleOptionClick}
        handleClose={mockHandleClose}
        handleDepositTypeChange={mockHandleDepositTypeChange}
        handleDepositAmountChange={mockHandleDepositAmountChange}
        handleDepositSubmit={mockHandleDepositSubmit}
        handleBuySubmit={mockHandleBuySubmit}
        handleBuyTypeChange={mockHandleBuyTypeChange}
        handleBuyAmountChange={mockHandleBuyAmountChange}
      />
    );
    fireEvent.click(getByRole('button', { name: /cerrar/i }));
    expect(mockHandleClose).toHaveBeenCalledTimes(1);
  });

  test('should have correct classes applied to elements', () => {
    const { container } = render(
      <OperationsDashboard 
        accountInfo={accountInfo}
        handleWithdrawTypeChange={mockHandleWithdrawTypeChange}
        handleWithdrawalAmountChange={mockHandleWithdrawalAmountChange}
        handleWithdrawalSubmit={mockHandleWithdrawalSubmit}
        state={state}
        selectedOption={selectedOption}
        handleOptionClick={mockHandleOptionClick}
        handleClose={mockHandleClose}
        handleDepositTypeChange={mockHandleDepositTypeChange}
        handleDepositAmountChange={mockHandleDepositAmountChange}
        handleDepositSubmit={mockHandleDepositSubmit}
        handleBuySubmit={mockHandleBuySubmit}
        handleBuyTypeChange={mockHandleBuyTypeChange}
        handleBuyAmountChange={mockHandleBuyAmountChange}
      />
    );
    expect(container.firstChild).toHaveClass('OperationsDashboard__wrapper');
  });

  test('should match snapshot', () => {
    const { asFragment } = render(
      <OperationsDashboard 
        accountInfo={accountInfo}
        handleWithdrawTypeChange={mockHandleWithdrawTypeChange}
        handleWithdrawalAmountChange={mockHandleWithdrawalAmountChange}
        handleWithdrawalSubmit={mockHandleWithdrawalSubmit}
        state={state}
        selectedOption={selectedOption}
        handleOptionClick={mockHandleOptionClick}
        handleClose={mockHandleClose}
        handleDepositTypeChange={mockHandleDepositTypeChange}
        handleDepositAmountChange={mockHandleDepositAmountChange}
        handleDepositSubmit={mockHandleDepositSubmit}
        handleBuySubmit={mockHandleBuySubmit}
        handleBuyTypeChange={mockHandleBuyTypeChange}
        handleBuyAmountChange={mockHandleBuyAmountChange}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
