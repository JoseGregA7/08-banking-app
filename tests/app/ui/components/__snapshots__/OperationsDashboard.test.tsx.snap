// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OperationsDashboard > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="OperationsDashboard__wrapper"
  >
    <div
      class="AccountStatus__wrapper"
    >
      <div
        class="AccountStatus__main"
      >
        <div
          class="AccountStatus__title"
        >
          Estado de la Cuenta
        </div>
        <div
          class="AccountStatus__content"
        >
          <ul
            class="AccountStatus__list"
          >
            <li
              class="AccountStatus__item"
            >
              <strong>
                balance:
              </strong>
               1000
            </li>
            <li
              class="AccountStatus__item"
            >
              <strong>
                currency:
              </strong>
               USD
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="Option__wrapper"
    >
      <div
        class="Option__title"
      >
        Opciones de la Cuenta
      </div>
      <div
        class="Option__container"
      >
        <button
          aria-label="option_button_dep贸sitos"
          class="Option__button"
          name="option_button_dep贸sitos"
          role="option_button_dep贸sitos"
        >
          Dep贸sitos
        </button>
        <button
          aria-label="option_button_retiros"
          class="Option__button"
          name="option_button_retiros"
          role="option_button_retiros"
        >
          Retiros
        </button>
        <button
          aria-label="option_button_compras"
          class="Option__button"
          name="option_button_compras"
          role="option_button_compras"
        >
          Compras
        </button>
      </div>
      <div
        class="Option__close"
      >
        <div>
          <button
            aria-label="Cerrar"
            name="cerrar"
            role="button"
          >
            x
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
